scrapy/utils/iterators.py

```diff
@@ -184,10 +181,13 @@ def csviter(
     quotechar is the character used to enclosure fields on the given obj.
     """
 
-    encoding = obj.encoding if isinstance(obj, TextResponse) else encoding or "utf-8"
-
-    def row_to_unicode(row_: Iterable) -> List[str]:
-        return [to_unicode(field, encoding) for field in row_]
+    if encoding is not None:
+        warn(
+            "The encoding argument of csviter() is ignored and will be removed"
+            " in a future Scrapy version.",
+            category=ScrapyDeprecationWarning,
+            stacklevel=2,
+        )
```

Comment:
Shouldn’t we keep the encoding functionality until we remove the parameter?

Reply:
The existing encoding functionality requires getting bytes in row_to_unicode(), which can't happen as all this code operates on strings (after `_body_or_str(obj, unicode=True)` converts the input with a hardcoded utf-8 encoding, anyway), so it was a no-op.

Reply:
AFAICS it was for the Py2 code path where `csv_r` produced bytes.